hack/pin-custom-bundle-dockerfile.sh

```diff
@@ -36,13 +36,14 @@ for MOD_PATH in $(go list -m -json all | jq -r '. | select(.Path | contains("ope
             REPO_CURL_URL="https://quay.io/api/v1/repository/${GITHUB_USER}"
             REPO_URL="quay.io/${GITHUB_USER}"
         fi
+        if [[ ${LOCAL_REGISTRY} -eq 1 ]]; then
+            SHA=$(curl -s ${REPO_CURL_URL}/$BASE-operator-bundle/tags/list | jq -r .tags[] | sort -u | grep $REF)
+        else
+            SHA=$(curl -s ${REPO_CURL_URL}/$BASE-operator-bundle/tag/ | jq -r .tags[].name | sort -u | grep $REF)
+        fi
     fi
 
-    if [[ ${LOCAL_REGISTRY} -eq 1 && "$BASE" == "$IMAGEBASE" ]]; then
```

Comment:
ok the base check is required 


the problem is basically that when using the content provider job we are using a local regiesty but then "BASE" == "$imageBASE" is not true for the nova operator.

and sha end ups being empty because we take the wrong path.

but if i move the around the it fails on the cinder operator because it is not using the local registry.

Reply:
perhapse instead i need to check "$GITHUB_USER" != "openstack-k8s-operators" ||

on the PR where the content provider job is running the $GITHUB_USER when processing the nova operator (since the pr was against nova) is actually my user or rather the user of the source of the pr rhater then the repo its being submitted too.

Reply:
if we look at the first run that got this far before i modified the code

https://review.rdoproject.org/zuul/build/bb2d1bddb52f49a9b9da2d343234aa29/log/ci-framework/logs/ci_make_011_openstack-operator_-_call_bundle.log#295-316

we can see it took the wrong branch and used the v1 API 
and that BASE=nova and GITHUB_USER=SeanMooney

we can also see where it check if we were uign the local regiesty
 if [[ ${LOCAL_REGISTRY} -eq 1 && "$BASE" == "$IMAGEBASE" ]]; then
+ [[ 1 -eq 1 ]]
+ [[ nova == '' ]]

so for some reason IMAGEBASE was not set

we can see it starts unset 

https://review.rdoproject.org/zuul/build/bb2d1bddb52f49a9b9da2d343234aa29/log/ci-framework/logs/ci_make_011_openstack-operator_-_call_bundle.log#23

and is never assigned too
so "$BASE" == "$IMAGEBASE" is not working its always false
so it need to be updated with something else.